dask/array/core.py

```diff
@@ -3757,12 +3757,27 @@ def concatenate3(arrays):
            [1, 2, 1, 2],
            [1, 2, 1, 2]])
     """
+    from .utils import IS_NEP18_ACTIVE
+    # We need this as __array_function__ may not exist on older NumPy versions.
+    # And to reduce verbosity.
+    NDARRAY_ARRAY_FUNCTION = getattr(np.ndarray, '__array_function__', None)
```

Comment:
This is fragile: on numpy 1.16 it is defined, but dispatching is turned off. In astropy, I do:
```
if NUMPY_LT_1_17:  # pragma: no cover
    # pre 1.16, overrides.py did not exist; in 1.16, it existed, but
    # __array_function__ overrides were turned off by default.
    ARRAY_FUNCTION_ENABLED = (hasattr(np.core, 'overrides') and
                              np.core.overrides.ENABLE_ARRAY_FUNCTION)
else:
    # In 1.17, overrides are enabled by default, but it is still possible to
    # turn them off using an environment variable.  We use getattr since it
    # is planned to remove that possibility in later numpy versions.
    ARRAY_FUNCTION_ENABLED = getattr(np.core.overrides,
                                     'ENABLE_ARRAY_FUNCTION', True)
```

Reply:
This isn't testing for `__array_function__`, that's in `IS_NEP18_ACTIVE`.

Reply:
@mhvk That environment variables uses the wrong internal variable name. I think it's `ARRAY_FUNCTION_ENABLED` now on NumPy master (though I would suggest just checking the environment variable instead).

Reply:
Duh, sorry for the noise.